src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -389,11 +389,11 @@ private void AddClassifiedSpansForTokens(IEditorClassificationService classifica
                     span.Snapshot.AsText(), span.Span.ToTextSpan(), classifiedSpans, CancellationToken.None);
             }
 
-            private void OnDocumentActiveContextChanged(object sender, DocumentEventArgs args)
+            private void OnDocumentActiveContextChanged(object sender, DocumentActiveContextChangedEventArgs args)
             {
-                if (_workspace != null)
+                if (_workspace != null && _workspace == args.Workspace)
                 {
-                    ParseIfThisDocument(null, args.Document.Project.Solution, args.Document.Id);
+                    ParseIfThisDocument(null, args.Workspace.CurrentSolution, args.NewContextDocumentId);
```

Comment:
its true that changing active document won't mutate solution in current implementation unlike open/close/workspace changed events.

but in perfect world, open/close document shouldn't be changing solution as well, but due to our implementation (associating different buffer to document), we do.

I think it is safer for us to include solution in active context events as well. also I think it is more consistent with other workspace events.